hpack/hpack.py

```diff
@@ -264,20 +155,20 @@ def add(self, to_add, sensitive, huffman=False):
         indexbit = INDEX_INCREMENTAL if not sensitive else INDEX_NEVER
 
         # Search for a matching header in the header table.
-        match = self.matching_header(name, value)
+        match = self.header_table.search(name, value)
 
         if match is None:
             # Not in the header table. Encode using the literal syntax,
             # and add it to the header table.
             encoded = self._encode_literal(name, value, indexbit, huffman)
             if not sensitive:
-                self._add_to_header_table(to_add)
+                self.header_table.add(name, value)
             return encoded
 
         # The header is in the table, break out the values. If we matched
         # perfectly, we can use the indexed representation: otherwise we
         # can use the indexed literal.
-        index, perfect = match
+        index, hname, perfect = match
```

Comment:
Well what I meant is, that hname value isn't worth anything in that context because it's always equal to the name value.  We do test this particular flow with the `test_indexed_literal_header_field_with_indexing` test : 

        e = Encoder()
        header_set = {':path': '/sample/path'}
        result = b'\x44\x0c/sample/path'

That header will be a partial match and dictionary header args have `sensitive` to `False` implicitly.  Unless I'm missing something.  There was nothing to catch I just thought that returning the header name on a partial match would be useful (I can't remember why now).